test_scripts/Defects/5_1/1002_SDL_sends_StopAudioStream_in_case_User_Exit_during_Audio_Video_streaming.lua

```diff
@@ -0,0 +1,107 @@
+-- 2) Navigation App is registered and activated
+-- 3) Video streaming is started
+-- 4) Audio streaming is started
+-- In case:
+-- 1) User Exit from App
+-- SDL does:
+-- send Navigation.OnAudioDataStreaming(available = false) notification to HMI
+-- send Navigation.OnVideoDataStreaming(available = false) notification to HMI
+-- send Navigation.StopAudioStream request to HMI
+-- send Navigation.StopStream request to HMI
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require("user_modules/sequences/actions")
+local events = require("events")
+local constants = require("protocol_handler/ford_protocol_constants")
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+config.defaultProtocolVersion = 3
+config.application1.registerAppInterfaceParams.appHMIType = { "NAVIGATION" }
+
+--[[ Local Variables ]]
+local audioServiceId = 10
+local videoServiceId = 11
+
+--[[ Local Functions ]]
+local function appStartVideoStreaming()
+  common.getMobileSession():StartService(videoServiceId)
+  :Do(function()
+      common.getHMIConnection():ExpectRequest("Navigation.StartStream")
+      :Do(function(_, data)
+          common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+          common.getMobileSession():StartStreaming(videoServiceId, "files/SampleVideo_5mb.mp4")
+          common.getHMIConnection():ExpectNotification("Navigation.OnVideoDataStreaming", { available = true })
+        end)
+    end)
+  common.getMobileSession():ExpectNotification("OnHMIStatus")
+  :Times(0)
+end
+
+local function appStartAudioStreaming()
+  common.getMobileSession():StartService(audioServiceId)
+  :Do(function()
+      common.getHMIConnection():ExpectRequest("Navigation.StartAudioStream")
+      :Do(function(_, data)
+          common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+          common.getMobileSession():StartStreaming(audioServiceId,"files/MP3_1140kb.mp3")
+          common.getHMIConnection():ExpectNotification("Navigation.OnAudioDataStreaming", { available = true })
+        end)
+    end)
+  common.getMobileSession():ExpectNotification("OnHMIStatus")
+  :Times(0)
+end
+
+local function expectEndService(pServiceId)
+  local event = events.Event()
+  event.matches = function(_, data)
+    return data.frameType == constants.FRAME_TYPE.CONTROL_FRAME
+    and data.serviceType == pServiceId
+    and data.sessionId == common.getMobileSession().mobile_session_impl.control_services.session.sessionId.get()
+    and data.frameInfo == constants.FRAME_INFO.END_SERVICE
+  end
+  return common.getMobileSession():ExpectEvent(event, "EndService")
+end
+
+local function endServiceByUserExit()
+  common.getHMIConnection():SendNotification("BasicCommunication.OnExitApplication",
+  { appID = common.app.getHMIId(), reason = "USER_EXIT" })
```

Comment:
For better stability I would move this action to the end of function (before any expectation)